modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ddl/DdlCommandHandler.java

```diff
@@ -237,33 +242,35 @@ private void handleCreateIndex(CreateIndexCommand cmd) {
                 IgniteObjectName.quote(cmd.tableName())
         );
 
-        tableManager.alterTable(fullName, chng -> chng.changeIndices(idxes -> {
-            if (idxes.get(cmd.indexName()) != null) {
-                if (!cmd.ifIndexNotExists()) {
-                    throw new IndexAlreadyExistsException(cmd.indexName());
-                } else {
-                    return;
-                }
+        try {
+            idxManager.createIndex(cmd.indexName(), fullName, idxCh -> convert(idx.build(), idxCh));
+        } catch (IndexAlreadyExistsException e) {
+            if (!cmd.ifIndexNotExists()) {
+                throw e;
```

Comment:
Looks like an exception is used for returning the 'result'.
Is it possible to pass a flag or better have a method createIndexIfNotExists ?

Reply:
I guess creation table & index must be handled in the same style.
The decision about tables is: IGNITE-15926 DDL methods for tables should throw an exception when the tables exist or not